discretisedfield/field.py

```diff
@@ -3153,6 +3204,54 @@ def _fromvtk(cls, filename):
 
         .. seealso:: :py:func:`~discretisedfield.Field._writevtk`
 
+        """
+        with open(filename, 'rb') as f:
+            xml = 'xml' in f.readline().decode('utf8')
+        if xml:
+            reader = vtk.vtkXMLRectilinearGridReader()
+        else:
+            reader = vtk.vtkRectilinearGridReader()
+            reader.ReadAllVectorsOn()
+            reader.ReadAllScalarsOn()
+        reader.SetFileName(filename)
+        reader.Update()
+
+        output = reader.GetOutput()
+        p1 = output.GetBounds()[::2]
+        p2 = output.GetBounds()[1::2]
+        n = [i - 1 for i in output.GetDimensions()]
+
+        cell_data = output.GetCellData()
+
+        if cell_data.GetNumberOfArrays() == 0:
+            # Old writing routine did write to points instead of cells.
+            return cls._fromvtk_legacy(filename)
+
+        components = []
+        for i in range(cell_data.GetNumberOfArrays()):
+            name = cell_data.GetArrayName(i)
+            if name == 'field':
+                field_idx = i
+            else:
+                components.append(name[:-len('-component')])
+        array = cell_data.GetArray(field_idx)
+        dim = array.GetNumberOfComponents()
+
+        if len(components) != dim:
+            components = None
+
+        value = vns.vtk_to_numpy(array).reshape(*reversed(n), dim)
+        value = value.transpose((2, 1, 0, 3))
+
+        mesh = df.Mesh(p1=p1, p2=p2, n=n)
+        return cls(mesh, dim=dim, value=value, components=components)
+
+    @classmethod
+    def _fromvtk_legacy(cls, filename):
```

Comment:
I would probably put this to `df.tools`.

Reply:
Everything in `df.tools` currently is public - which this function should never be. It should always only be used automatically if we detect that we cannot read the file with the new function. If moving it at all I think `df.util` would be a better place.

Reply:
I was thinking like... you tried to open a VTK file that was written a long time ago... you cannot and that functionality is now deprecated. However, you can find a "tool" in "df.tools" that helps you with that. I would really like to avoid cluttering Field class, especially because the chances of anyone needing this function is infinitesimally small.

Reply:
Then maybe we can just move it to `df.util`? That way we could still use it automatically (which is only two lines of code) (and the import is already there) without the need for the user to do anything manually. Explaining to users (or even a deprecation warning) would be more lines (of code) that just using the function from `df.util`.